chapter-05/final/money/decimal.go

```diff
@@ -0,0 +1,128 @@
+package money
+
+import (
+	"fmt"
+	"math"
+	"strconv"
+	"strings"
+)
+
+// number is a structure that can hold a number with a fixed precision.
+// example: 1.52 = 1 + 52 * 10^(-2) will be stored as {1, 52, 2}
+type number struct {
+	// the integer part of the number
+	integerPart int
+	// the decimal part of the number
+	decimalPart int
+	// precision of the decimal part, as the exponent of a power of 10
+	precision int
+}
+
+const (
+	// errors that we might face when parsing numbers
+	errInvalidInteger = moneyError("unable to convert integer part")
+	errInvalidDecimal = moneyError("unable to convert decimal part")
```

Comment:
I don't think there's a use for having separate sentinel errors, having appropriate text would be fine.

Reply:
It might be overengineered but we take the opportunity to teach it in this chapter and unload the following chapter.